Crash when module source is incorrect

### Terraform Version

Terraform v0.11.2

### Terraform Configuration Files
```hcl
provider "aws" {}
module "http_sg" {
  source = "terraform-aws-modules/security-group/aws//modules/http-80"
}
```

### Debug Output
```
$ terraform init
Initializing modules...
- module.http_sg
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1521f06]

goroutine 1 [running]:
github.com/hashicorp/terraform/registry.(*Client).Discover(0xc4203ef180, 0x3343af9, 0x15, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/registry/client.go:70 +0x66
github.com/hashicorp/terraform/registry.(*Client).Versions(0xc4203ef180, 0xc42038edc0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/registry/client.go:83 +0xa2
github.com/hashicorp/terraform/config/module.Storage.findRegistryModule(0xc42034ebc0, 0x12, 0x0, 0x0, 0x0, 0x4abbe00, 0xc420369da0, 0x1, 0xc4203ef180, 0xc420019f81, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/storage.go:319 +0x6eb
github.com/hashicorp/terraform/config/module.(*Tree).getChildren(0xc420128f00, 0xc42038ed20, 0xc420128f38, 0xc4203ef180, 0xc4203ef180)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:219 +0x493
github.com/hashicorp/terraform/config/module.(*Tree).Load(0xc420128f00, 0xc42038ed20, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:178 +0xab
github.com/hashicorp/terraform/command.getModules(0xc42016ca80, 0xc420018104, 0x2e, 0xc420017b01, 0x23, 0xc420017b60)
	/opt/gopath/src/github.com/hashicorp/terraform/command/get.go:84 +0x135
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc42016ca80, 0xc42000a090, 0x0, 0x0, 0xc42033cd00)
	/opt/gopath/src/github.com/hashicorp/terraform/command/init.go:178 +0xb50
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc4202cfcc0, 0xc4202cfcc0, 0xc4201d7d18, 0x1)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:255 +0x1eb
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:214 +0xbda
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:99 +0xbb
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:35 +0x38
```

### Crash Output
https://gist.github.com/antonbabenko/11de9aecb65ba369245011bb2b843f63

### Expected Behavior
Do not panic. Nicer error message.

### Actual Behavior
Panic.

### Steps to Reproduce
1. `terraform init`

--- a/registry/client.go
+++ b/registry/client.go
@@ -67,6 +67,9 @@ func NewClient(services *disco.Disco, creds auth.CredentialsSource, client *http
 // Discover qeuries the host, and returns the url for the registry.
 func (c *Client) Discover(host svchost.Hostname) *url.URL {
 	service := c.services.DiscoverServiceURL(host, serviceID)
+	if service == nil {
+		return nil
+	}
 	if !strings.HasSuffix(service.Path, "/") {
 		service.Path += "/"
 	}

--- a/registry/client_test.go
+++ b/registry/client_test.go
@@ -55,6 +55,23 @@ func TestLookupModuleVersions(t *testing.T) {
 	}
 }
 
+func TestInvalidRegistry(t *testing.T) {
+	server := test.Registry()
+	defer server.Close()
+
+	client := NewClient(test.Disco(server), nil, nil)
+
+	src := "non-existent.localhost.localdomain/test-versions/name/provider"
+	modsrc, err := regsrc.ParseModuleSource(src)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := client.Versions(modsrc); err == nil {
+		t.Fatal("expected error")
+	}
+}
+
 func TestRegistryAuth(t *testing.T) {
 	server := test.Registry()
 	defer server.Close()

